src/game.cpp

```diff
@@ -578,54 +629,193 @@ void test_game() //!OCLINT tests may be many
+    g.set_player_state_passive(g.get_player(0));
+    g.remove_player_state_passive(g.get_player(0));
+    assert(!is_passive(g.get_player(0)));
+  }
+
+  /// A player can become non-passive again, using player.remove_state_passive()
+  {
+    game g;
 
+    assert(is_active(g.get_player(0)));
+    g.get_player(0).set_state_passive();
+    g.get_player(0).remove_state_passive();
+    assert(!is_passive(g.get_player(0)));
   }
   #endif
 
+  #define FIX_ISSUE_382
   #ifdef FIX_ISSUE_382
-  ///An invulnerable player cannot shrink
+  /// An passive player cannot grow
   {
     game g;
 
-    //Make the first player invulnerable
-    become_invulnerable(g.get_player(0));
+    // Make the first player passive
+    g.set_player_state_passive(g.get_player(0));
 
     // Make player 1 and 2 overlap
     g.get_player(1).set_x(get_x(g.get_player(0)));
     g.get_player(1).set_y(get_y(g.get_player(0)));
-    assert(has_collision(g));
 
-    // After a tick, invulnerable player does not shrink
-    const int inv_player_size_before =  get_nth_player_size(g, 0);
+    const int first_player_index = get_collision_members(g)[0];
+    const int second_player_index = get_collision_members(g)[1];
+    const int winner_index = get_winning_player_index(g, first_player_index, second_player_index);
+
+    assert(winner_index == 0);
+
+    // After a tick, passive player does not grow
+    const int inv_player_size_before =  get_nth_player_size(g, winner_index);
     g.tick();
-    const int inv_player_size_after =  get_nth_player_size(g, 0);
+    const int inv_player_size_after =  get_nth_player_size(g, winner_index);
+    assert(inv_player_size_after == inv_player_size_before);
+  }
+
+  /// An passive player cannot shrink
+  {
+    game g;
+
+    //Make the first player passive
+    g.set_player_state_passive(g.get_player(1));
+
+    // Make player 1 and 2 overlap
+    g.get_player(1).set_x(get_x(g.get_player(0)));
+    g.get_player(1).set_y(get_y(g.get_player(0)));
+
+    const int first_player_index = get_collision_members(g)[1];
+    const int second_player_index = get_collision_members(g)[0];
+    const int loser_index = get_losing_player_index(g, first_player_index, second_player_index);
+
+    assert(loser_index == 1);
+
+    // After a tick, passive player does not shrink
+    const int inv_player_size_before =  get_nth_player_size(g, loser_index);
+    g.tick();
+    const int inv_player_size_after =  get_nth_player_size(g, loser_index);
     assert(inv_player_size_after == inv_player_size_before);
   }
   #endif
 
-  //#define FIX_ISSUE_463
+
+  #define FIX_ISSUE_463
   #ifdef FIX_ISSUE_463
-  // Players lose invulnerability after a short time
+  // Can get and set a player's passive state
+  {
+    game g;
+    player& p = g.get_player(0);
+
+    // Check default duration
+    assert(p.get_duration_passive() == 2000);
+
+    int new_duration = 200;
+    p.set_duration_passive(new_duration);
+    assert(p.get_duration_passive() == new_duration);
+  }
```

Comment:
I can't see why you would change the duration of the passive state at the player level; shouldn't it be a fixed parameter of the game?

Reply:
@TheoPannetier I was thinking about possible future extension of the power-ups etc. For a minimal viable game, it should also be good to just let the duration be a fixed parameter of the game. What is your opinion about this?

Reply:
I think you've figured out my opinion already 😉 
I believe anticipating possible extensions beyond the minimal viable game is premature optimisation, and a wise man once told me it's the [root of all evil](https://effectiviology.com/premature-optimization/).

So for now, I'd rather have the duration as a fixed parameter. It will be easy to make it free again when/if we need it :)

Reply:
> I think you've figured out my opinion already 😉 I believe anticipating possible extensions beyond the minimal viable game is premature optimisation, and a wise man once told me it's the [root of all evil](https://effectiviology.com/premature-optimization/).
> 
> So for now, I'd rather have the duration as a fixed parameter. It will be easy to make it free again when/if we need it :)

I'll change it in the following commits